### PR TITLE
feat-documentation

### PR DESCRIPTION
**Pull request by KevinMB0220**
### 📝 Summary 
- Closes #(issue)
- 
### 🔄 Changes Made
- 
### 🔧 Tests Results
- 

### 🔜 Next Steps
- 
PoC inicial

### CODE REVIEW

File: src/systems/battle.cairo [hunk 1]
```diff
@@ -61,117 +93,150 @@ mod battle_system {
...
+        /// Checks if the player can flee based on beast levels.
         fn check_flee_success(player_beast: Beast, opponent_beast: Beast) -> bool {
             player_beast.level > opponent_beast.level 
         }  
 
+         /// Calculates the damage dealt by an attacker to a defender.
         fn calculate_damage(mt: Mt, attacker: Beast, defender: Beast) -> u32 {
-            let base_damage = mt.mt_power * attacker.attack / defender.defense;
-
-            //TODO: extend with effectivity by type, randomness, etc
```
Comment (coxmars) [inline src/systems/battle.cairo:78]: @KevinMB0220, please keep the existing comments as they serve as a guide for future improvements in specific sections. You can add any new comments alongside them.

File: src/systems/battle.cairo [hunk 2]
```diff
@@ -61,117 +93,150 @@ mod battle_system {
...
-            let effective_damage = base_damage; 
+            let base_damage = mt.mt_power * attacker.attack / defender.defense;  // Basic damage formula
 
-            // TODO: investigate how can we make It random
```
Comment (coxmars) [inline src/systems/battle.cairo:81]: Same applies here

File: src/models/battle.cairo [hunk 3]
```diff
@@ -1,22 +1,52 @@
+// ***************************************************************
+//                           MODEL DEFINITION
+// ***************************************************************
+
+/// Defines the `Battle` struct, which represents a battle between two players.
```
Comment (danielcdz) [inline src/models/battle.cairo:5]: can you replace the word `struct` to `model`?

File: src/models/battle.cairo [hunk 4]
```diff
@@ -1,22 +1,52 @@
...
+// ***************************************************************
+
 #[cfg(test)]
 mod tests {
+    // Imports the `Battle` struct from the `bytebeasts` models.
     use bytebeasts::{models::{battle::Battle},};
 
+    // ***************************************************************
+    //                   TEST CASE: Battle Initialization
+    // ***************************************************************
```
Comment (danielcdz) [inline src/models/battle.cairo:44]: Can you move to the top of the test? place it in line 40, in most of this tests for models we do testing for the initialization of the model, so I think is better to leave it on the top level

File: src/models/battle.cairo [hunk 5]
```diff
@@ -1,22 +1,52 @@
...
     pub turn: u32,
 }
 
+// ***************************************************************
+//                           TEST MODULE
+// ***************************************************************
+
 #[cfg(test)]
 mod tests {
+    // Imports the `Battle` struct from the `bytebeasts` models.
```
Comment (danielcdz) [inline src/models/battle.cairo:39]: for imports we can avoid the comments since are very intuitive

File: src/models/battle.cairo [hunk 6]
```diff
@@ -27,6 +57,7 @@ mod tests {
             turn: 1,
         };
 
+        // Asserts to check if the `battle` instance has the expected values.
```
Comment (danielcdz) [inline src/models/battle.cairo:60]: this also can be avoided

File: src/models/battle.cairo [hunk 7]
```diff
@@ -1,22 +1,52 @@
...
     use bytebeasts::{models::{battle::Battle},};
 
+    // ***************************************************************
+    //                   TEST CASE: Battle Initialization
+    // ***************************************************************
+
+    /// Test to verify the correct initialization of a `Battle` instance.
     #[test]
     fn test_battle_initialization() {
+        // Creates an instance of `Battle` with predefined values for testing.
```
Comment (danielcdz) [inline src/models/battle.cairo:49]: this can be avoided

File: src/systems/battle.cairo [hunk 8]
```diff
@@ -1,52 +1,84 @@
...
-    fn attack(ref world: IWorldDispatcher, battle_id: u32, mt_id: u32);
-    fn use_potion(ref world: IWorldDispatcher, battle_id: u32, potion_id: u32);
-    fn flee(ref world: IWorldDispatcher, battle_id: u32);
+    fn init_battle(ref world: IWorldDispatcher, player_id: u32, opponent_id: u32) -> u32; // Initializes a battle
+    fn check_flee_success(player_beast: Beast, opponent_beast: Beast) -> bool; // Checks flee success
+    fn calculate_damage(mt: Mt, attacker: Beast, defender: Beast) -> u32; // Calculates damage
+    fn opponent_turn(ref world: IWorldDispatcher, battle_id: u32); // Handles opponent's turn
+    fn attack(ref world: IWorldDispatcher, battle_id: u32, mt_id: u32); // Executes an attack
+    fn use_potion(ref world: IWorldDispatcher, battle_id: u32, potion_id: u32); // Uses a potion
+    fn flee(ref world: IWorldDispatcher, battle_id: u32); // Attempts to flee
```
Comment (danielcdz) [inline src/systems/battle.cairo:23]: Can you change this to be on top of the function definition, for example:

``` rust
    // Executes an attack
    fn attack(ref world: IWorldDispatcher, battle_id: u32, mt_id: u32); 
   // Uses a potion
    fn use_potion(ref world: IWorldDispatcher, battle_id: u32, potion_id: u32); 

```

File: src/systems/battle.cairo [hunk 9]
```diff
@@ -1,52 +1,84 @@
...
+//                           CONTRACT MODULE
+// ***************************************************************
+
+/// Contract implementing battle actions.
 #[dojo::contract]
 mod battle_system {
-    use super::{IBattleActions};
+// ***************************************************************
+//                           IMPORTS
+// ***************************************************************
```
Comment (danielcdz) [inline src/systems/battle.cairo:35]: this can be avoided

**Review by danielcdz — CHANGES_REQUESTED** [review]

**Comment by danielcdz** [comment]
In some comments you are using `///` and in other `//` can you update all of them to be just `//`?

**Comment by KevinMB0220** [comment]
yes, I'm almost done pr @danielcdz

**Pull request closed without merging**